hazelcast/src/main/java/com/hazelcast/topic/impl/reliable/MessageRunner.java

```diff
@@ -113,14 +118,11 @@ public void accept(ReadResultSet<ReliableTopicMessage> result, Throwable throwab
                     }
                 }
 
-                sequence++;
             }
+
+            sequence = result.getNextSequenceToReadFrom();
```

Comment:
you update the member `sequence` only after the iteration is finished but if the exception occurs and it is a terminal exception, the method returns leaving the runner with incorrect sequence number. The master branch code updates the sequence during iteration. Should we revert this back?

Reply:
If the iteration is aborted for some reason, then the runner is also cancelled, so it doesn't matter what the sequence end up like. At least that's how I see it. Am I mistaken?

Reply:
If `terminate(t)` is false, it will not abort, and continue, right?

Reply:
If `terminate(t)` is `false` then it will go on with the `for` loop and once the loop is done it will set the `sequence` correctly. `sequence` is no longer used anywhere inside the loop, so this should be fine, right?

Reply:
Seems fine to me.